ovos_plugin_manager/templates/agent_tools.py

```
@@ -0,0 +1,310 @@
+        if not tool:
+            raise ValueError(f"Unknown tool '{name}' for ToolBox '{self.toolbox_id}'.")
+
+        # 1. Input Validation and Instantiation
+        if isinstance(tool_kwargs, ToolArguments):
+            # Case A: Input is an already validated Pydantic model.
+            # We perform a quick type check to ensure it matches the declared schema.
+            if not isinstance(tool_kwargs, tool.argument_schema):
+                raise ValueError(
+                    f"Tool '{name}' called with model of type {type(tool_kwargs).__name__}, "
+                    f"but expected {tool.output_schema.__name__}."
+                )
```

Ruff (0.14.0)

Prefer `TypeError` exception for invalid type

(TRY004)